The coveralls job is broken in continuous integration

See [here](https://github.com/PyCQA/pylint/pull/4401/checks?check_run_id=2437252161)

```
Submitting coverage to coveralls.io...
Could not submit coverage: 500 Server Error: Internal Server Error for url: https://coveralls.io/api/v1/jobs
Traceback (most recent call last):
  File "/home/runner/work/pylint/pylint/venv/lib/python3.8/site-packages/coveralls/api.py", line 257, in wear
    response.raise_for_status()
  File "/home/runner/work/pylint/pylint/venv/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://coveralls.io/api/v1/jobs

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/pylint/pylint/venv/lib/python3.8/site-packages/coveralls/cli.py", line 85, in main
    result = coverallz.wear()
  File "/home/runner/work/pylint/pylint/venv/lib/python3.8/site-packages/coveralls/api.py", line 260, in wear
    raise CoverallsException(
coveralls.exception.CoverallsException: Could not submit coverage: 500 Server Error: Internal Server Error for url: https://coveralls.io/api/v1/jobs
Error: Process completed with exit code 1.
```

@cdce8p do you have an idea how to fix it ?

--- a/ChangeLog
+++ b/ChangeLog
@@ -149,6 +149,9 @@ Release date: TBA
 
   Closes #4324
 
+* Added the ``super-without-brackets`` checker, raised when a super call is missing its brackets.
+
+  Closes #4008
 
 * Added the ``generate-toml-config`` option.
 

--- a/doc/data/messages/s/super-without-brackets/bad.py
+++ b/doc/data/messages/s/super-without-brackets/bad.py
@@ -1,0 +1,11 @@
+class Soup:
+    @staticmethod
+    def temp():
+        print("Soup is hot!")
+
+
+class TomatoSoup(Soup):
+    @staticmethod
+    def temp():
+        super.temp()  # [super-without-brackets]
+        print("But tomato soup is even hotter!")

--- a/doc/data/messages/s/super-without-brackets/good.py
+++ b/doc/data/messages/s/super-without-brackets/good.py
@@ -1,0 +1,11 @@
+class Soup:
+    @staticmethod
+    def temp():
+        print("Soup is hot!")
+
+
+class TomatoSoup(Soup):
+    @staticmethod
+    def temp():
+        super().temp()
+        print("But tomato soup is even hotter!")

--- a/doc/whatsnew/2.14.rst
+++ b/doc/whatsnew/2.14.rst
@@ -44,6 +44,10 @@ New checkers
 
   Closes #5880
 
+* Added the ``super-without-brackets`` checker, raised when a super call is missing its brackets.
+
+  Closes #4008
+
 Removed checkers
 ================
 

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -38,7 +38,7 @@ from pylint.checkers.utils import (
     unimplemented_abstract_methods,
     uninferable_final_decorators,
 )
-from pylint.interfaces import INFERENCE, IAstroidChecker
+from pylint.interfaces import HIGH, INFERENCE, IAstroidChecker
 
 if sys.version_info >= (3, 8):
     from functools import cached_property
@@ -1523,7 +1523,7 @@ a metaclass class method.",
             return
 
         if node.expr.name == "super":
-            self.add_message("super-without-brackets", node=node.expr)
+            self.add_message("super-without-brackets", node=node.expr, confidence=HIGH)
 
     @check_messages(
         "assigning-non-slot", "invalid-class-object", "access-member-before-definition"

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -581,6 +581,12 @@ MSGS = {  # pylint: disable=consider-using-namedtuple-or-dataclass
         "redefined-slots-in-subclass",
         "Used when a slot is re-defined in a subclass.",
     ),
+    "W0245": (
+        "Super call without brackets",
+        "super-without-brackets",
+        "Used when a call to super does not have brackets and thus is not an actual "
+        "call and does not work as expected.",
+    ),
     "E0236": (
         "Invalid object %r in __slots__, must contain only non empty strings",
         "invalid-slots-object",
@@ -1490,6 +1496,8 @@ a metaclass class method.",
         class member from outside its class (but ignore __special__
         methods)
         """
+        self._check_super_without_brackets(node)
+
         # Check self
         if self._uses_mandatory_method_param(node):
             self._accessed.set_accessed(node)
@@ -1498,6 +1506,24 @@ a metaclass class method.",
             return
 
         self._check_protected_attribute_access(node)
+
+    def _check_super_without_brackets(self, node: nodes.Attribute) -> None:
+        """Check if there is a function call on a super call without brackets."""
+        # Check if attribute call is in frame definition in class definition
+        frame = node.frame()
+        if not isinstance(frame, nodes.FunctionDef):
+            return
+        if not isinstance(frame.parent.frame(), nodes.ClassDef):
+            return
+
+        if not isinstance(node.parent, nodes.Call):
+            return
+
+        if not isinstance(node.expr, nodes.Name):
+            return
+
+        if node.expr.name == "super":
+            self.add_message("super-without-brackets", node=node.expr)
 
     @check_messages(
         "assigning-non-slot", "invalid-class-object", "access-member-before-definition"

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -1515,13 +1515,10 @@ a metaclass class method.",
             return
         if not isinstance(frame.parent.frame(), nodes.ClassDef):
             return
-
         if not isinstance(node.parent, nodes.Call):
             return
-
         if not isinstance(node.expr, nodes.Name):
             return
-
         if node.expr.name == "super":
             self.add_message("super-without-brackets", node=node.expr, confidence=HIGH)
 

--- a/tests/functional/s/super/super_without_brackets.py
+++ b/tests/functional/s/super/super_without_brackets.py
@@ -1,0 +1,23 @@
+"""Tests for super-without-brackets"""
+
+# pylint: disable=missing-function-docstring, missing-class-docstring, too-few-public-methods
+
+
+class Soup:
+    @staticmethod
+    def temp():
+        print("Soup is hot!")
+
+
+class TomatoSoup(Soup):
+    @staticmethod
+    def temp():
+        super.temp()  # [super-without-brackets]
+        print("But tomato soup is even hotter!")
+
+
+class VegetableSoup(Soup):
+    @staticmethod
+    def temp():
+        super.temp()  # [super-without-brackets]
+        print("But tomato soup is even hotter!")

--- a/tests/functional/s/super/super_without_brackets.py
+++ b/tests/functional/s/super/super_without_brackets.py
@@ -19,5 +19,5 @@ class TomatoSoup(Soup):
 class VegetableSoup(Soup):
     @staticmethod
     def temp():
-        super.temp()  # [super-without-brackets]
-        print("But tomato soup is even hotter!")
+        super().temp()
+        print("But vegetable soup is even hotter!")

--- a/tests/functional/s/super/super_without_brackets.txt
+++ b/tests/functional/s/super/super_without_brackets.txt
@@ -1,0 +1,2 @@
+super-without-brackets:15:8:15:13:TomatoSoup.temp:Super call without brackets:UNDEFINED
+super-without-brackets:22:8:22:13:VegetableSoup.temp:Super call without brackets:UNDEFINED

--- a/tests/functional/s/super/super_without_brackets.txt
+++ b/tests/functional/s/super/super_without_brackets.txt
@@ -1,2 +1,1 @@
-super-without-brackets:15:8:15:13:TomatoSoup.temp:Super call without brackets:UNDEFINED
-super-without-brackets:22:8:22:13:VegetableSoup.temp:Super call without brackets:UNDEFINED
+super-without-brackets:15:8:15:13:TomatoSoup.temp:Super call without brackets:HIGH

